notebooks/03-query/20-tiering-historicals.ipynb

```diff
@@ -0,0 +1,790 @@
+   "source": [
+    "By setting up a table-specific rule set, where `tieredReplicants` includes both tiers, both the `slow` and `_default_tier` tiers have been loaded with all the segments of your table.\n",
+    "\n",
+    "(Run the cell above again if you do not see this immediately.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7a0ad47b-c6f9-45af-9278-4c81a1aaf6bc",
+   "metadata": {},
+   "source": [
+    "### Load tiers according to data age\n",
+    "\n",
+    "Run the next cell to create some \"fake\" data in the table that is a year older.\n",
+    "\n",
+    "* INSERT is used to append data instead of REPLACE INTO.\n",
+    "* The [TIME_PARSE](https://druid.apache.org/docs/latest/querying/sql-scalar#date-and-time-functions) function has been used to shift the timestamp back by a year. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dbd90368-7da5-4d2c-94b2-1bac4001291a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "INSERT INTO \"''' + table_name + '''\"\n",
+    "WITH \"ext\" AS (\n",
+    "  SELECT *\n",
+    "  FROM TABLE(\n",
+    "    EXTERN(\n",
+    "      '{\"type\":\"http\",\"uris\":[\"https://druid.apache.org/data/wikipedia.json.gz\"]}',\n",
+    "      '{\"type\":\"json\"}'\n",
+    "    )\n",
+    "  ) EXTEND (\"isRobot\" VARCHAR, \"channel\" VARCHAR, \"timestamp\" VARCHAR, \"flags\" VARCHAR, \"isUnpatrolled\" VARCHAR, \"page\" VARCHAR, \"diffUrl\" VARCHAR, \"added\" BIGINT, \"comment\" VARCHAR, \"commentLength\" BIGINT, \"isNew\" VARCHAR, \"isMinor\" VARCHAR, \"delta\" BIGINT, \"isAnonymous\" VARCHAR, \"user\" VARCHAR, \"deltaBucket\" BIGINT, \"deleted\" BIGINT, \"namespace\" VARCHAR, \"cityName\" VARCHAR, \"countryName\" VARCHAR, \"regionIsoCode\" VARCHAR, \"metroCode\" BIGINT, \"countryIsoCode\" VARCHAR, \"regionName\" VARCHAR)\n",
+    ")\n",
+    "SELECT\n",
+    "  TIME_SHIFT(TIME_PARSE(\"timestamp\"), 'P1Y', -1) AS \"__time\",\n",
+    "  \"isRobot\",\n",
+    "  \"channel\",\n",
+    "  \"isUnpatrolled\",\n",
+    "  \"page\",\n",
+    "  \"comment\",\n",
+    "  \"commentLength\",\n",
+    "  \"user\"\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "\n",
+    "display.run_task(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4dbdcc30-f0b1-494e-85bc-ea899a920f74",
+   "metadata": {},
+   "source": [
+    "Review the following retention rule.\n",
+    "\n",
+    "- An [interval load rule](https://druid.apache.org/docs/latest/operations/rule-configuration#interval-load-rule) (`loadByInterval`) covers 10 years of data before 1st January 2016, and requests one cached replica on both the `_default_tier` and `slow` tiers.\n",
+    "- `loadForever` requests that all data be cached on the `slow` tier.\n",
+    "\n",
+    "Each segment is checked against the rules in order when the decision is made as to where it must be cached.\n",
+    "\n",
+    "Run the cell to commit it to the database."
```

Comment:
This needs revision. The code accomplishes the correct task, but based upon this description I was confounded.

Discrepencies:

Line 431 directly opposes line 490, which correctly describes the result:
> Data for 2015 (`loadByInterval`) is only cached on the `slow` tier.

Line 432 directly opposes Line 491, which correctly describes the result:
> All other data (`loadForever`) is only available on the `_default_tier` tier.

* Need to say that Druid applies the load rules in order first
* Need to say that the interval load rule assigns the interval of 10 years up to 1 Jan 2016 to the _slow_ tier
* Need to say that somehow the remaining data gets assigned to the default tier based upon the loadForever rule. Like if data is already handled by the interval rule, it won't be assigned here.

Reply:
THANK YOU! I didn't spot that! Updated.